ui/app/actions.js

```diff
@@ -1270,8 +1270,10 @@ function retryTransaction (txId) {
       if (err) {
         return dispatch(actions.displayWarning(err.message))
       }
+      const { selectedAddressTxList } = newState
+      const { id: newTxId } =  selectedAddressTxList[selectedAddressTxList.length - 1]
```

Comment:
Oh hey, I don't know this form of restructuring. Can you describe what it does?

Reply:
It is equivalent to `const newTxId =  selectedAddressTxList[selectedAddressTxList.length - 1].id`

`const { propName: newConstName } = someObject` declares a new `const` called `newConstName` and points it to whatever the `propName` property of `someObject` is pointing to. If `someObject` does not have a `propName` property, then `newConstName === undefined`

Reply:
Not particularly useful in this instance, but in other cases it can help write self-documenting code while keeping code concise. For instance: https://github.com/MetaMask/metamask-extension/pull/3489/files#diff-0cd237b3791741831976770fdf19a000R77